### ISSUE TITLE
`plus/indent-binary-ops` expects wrong indentation with `typeof` operator

### ISSUE THREAD

**Issue by Ky6uk** [eslint-stylistic/eslint-stylistic#222]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

It looks like `plus/indent-binary-ops` works wrongly with `typeof` operator.

### Errors reported by the rule

<img width="656" alt="Screenshot 2023-12-11 at 21 29 17" src="https://github.com/eslint-stylistic/eslint-stylistic/assets/46930/64a7d9d1-ac1d-425a-b3c4-877de3dd0da6">

### The code if all formatting errors are fixed

<img width="359" alt="Screenshot 2023-12-11 at 21 30 07" src="https://github.com/eslint-stylistic/eslint-stylistic/assets/46930/6c498c28-a560-416e-bb89-7f097c4e7d5a">

### Reproduction

https://stackblitz.com/edit/eslint-stylistic-eslint-stylistic-q85j5j?file=playground%2Findex.js

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by mdziekon** [comment]
I've stumbled upon the same problem. As a temporary fix, looks like the `typeof` expression can be wrapped in parenthesis, which seems to solve the problem (and maybe will give a hint to someone working on this problem?), see:
https://stackblitz.com/edit/eslint-stylistic-eslint-stylistic-bcxv9k?file=playground%2Findex.js

```typescript
// This seems to work fine
if (
  (typeof woof === 'string') &&
  (typeof woof === 'string') &&
  (typeof woof === 'string') &&
  isNaN(null) &&
  isNaN(NaN)
) {
  return;
}
```
